### ISSUE TITLE
Introduce hard limit on transaction fee (legacy)

### ISSUE THREAD

**Issue by hynek-jina** [trezor/trezor-firmware#2191]
Let's implement this for Legacy too.

_Originally posted by @prusnak in https://github.com/trezor/trezor-firmware/issues/1087#issuecomment-1073721381_

**Comment by prusnak** [comment]
Why is this marked with Product label? We already discussed this at Product we want this.

**Comment by prusnak** [comment]
I don't think this needs the Product label or does it? It's been decided we want this.

**Comment by matejcik** [comment]
"product" does not mean "needs product input", it is an issue class category
